[RFR] Bootstrap the project

Luwangel opened the pull request:
- [x] Create the readme
- [x] Build an initial setup with unit tests
- [x] Create a dockerfile

Luwangel committed:
Bootstrap the project

Kmaschta commented on .travis.yml:
```diff
@@ -0,0 +1,20 @@
+language: go
+
+go:
+  - 1.7.3
```
Can you use the latest version of Go ?

Kmaschta commented on Dockerfile:
```diff
@@ -0,0 +1,6 @@
+FROM golang:1.8
```
Please use the latest version

Kmaschta commented on README.md:
```diff
@@ -1,2 +1,45 @@
+
+### Help
+
+Print all commands available
```
s/all commands available/all available commands/

sedy-bot committed:
Typo fix s/all commands available/all available commands/

As requested by Kmaschta at https://github.com/marmelab/15-puzzle-go/pull/1#discussion_r145665820

jpetitcolas commented on Dockerfile:
```diff
@@ -0,0 +1,6 @@
+
+ENV GOPATH /src
+
+VOLUME /src
```
Useless. Remove to keep it simple.

jpetitcolas commented on Makefile:
```diff
@@ -0,0 +1,17 @@
+PWD = $(shell pwd)
```
There is already a environment variable `${PWD}`.

Luwangel committed:
Use the latest go version and fix the gopath in docker

Luwangel committed:
Don't create an extra pwd variable
